### PR TITLE
Adding Sdk integration, and formatting support + code action for syntax updates to the language server

### PR DESCRIPTION
**Pull request by bettinaheim**
This PR does three things:
- It includes the published version of the formatter in the Microsoft.Quantum.Sdk, and adds a build target for updating the syntax in the project (no formatting).
- It adds support for formatting files in the language server. This currently as written includes both formatting and updating syntax.
- It adds a code action for the things that can be fixed by the syntax updater. If that action is selected, the deprecated syntax in the whole file is updates. No formatting is performed. 

There is a test for the language server functionality, which tests two scenarios:
1) The first part checks formatting works when using the formatter packaged in the Sdk, and relies on a fixed version of the Sdk that has the formatter included. 
2) The second part tests formatting works when using the (published) source code version of the formatter, and should catch any breaking changes to the formatter.

Additionally, we will set up a sanity test in the qdk repo to ensure that the formatter is correctly packaged in all future versions of the Sdk and the defined build target works as well. 
Telemetry will follow in a separate PR.

### CODE REVIEW

**Review by ScottCarda-MS — COMMENTED** [review]
The work in the formatter looks fine, as well as the changes around the ProjectProperties type. 

An updated version of the formatter documentation is merged into main. Please pull that in and update that with anything relevant to the changes in this PR. Off the top of my head, it could be very informative to add a new section about the three new ways this PR exposes the formatter for use to people through the SDK. I also believe you had mentioned something before about updating the 'Limitations' section.

**Review by ScottCarda-MS — APPROVED** [review]

File: src/QsCompiler/LanguageServer/LanguageServer.cs [hunk 1]
```diff
@@ -681,9 +700,7 @@ CodeAction CreateAction(string title, WorkspaceEdit edit)
             {
                 return
                     QsCompilerError.RaiseOnFailure(
-                        () => this.editorState.CodeActions(param)
-                            ?.SelectMany(vs => vs.Select(v => CreateAction(vs.Key, v)))
-                            ?? Enumerable.Empty<CodeAction>(),
+                        () => this.editorState.CodeActions(param) ?? Enumerable.Empty<CodeAction>(),
```
Comment (ricardo-espinoza) [inline src/QsCompiler/LanguageServer/LanguageServer.cs:703]: Ah, this is so much cleaner. :)

**Comment by ricardo-espinoza** [comment]
Thanks @bettinaheim for the detailed walk through this pull request. It took me a while to go over it, and catch up with the meeting last week, but looks good to me.

**Review by ricardo-espinoza — APPROVED** [review]

**Comment by bettinaheim** [comment]
@ScottCarda-MS I updated the readme of the formatter. Please take a look.

**Review by ScottCarda-MS — APPROVED** [review]
Thanks @bettinaheim. The README changes look good to me.

**Pull request merged**